codalab/worker/rest_client.py

```diff
@@ -135,32 +134,12 @@ def _upload_with_chunked_encoding(
         else:
             conn = http.client.HTTPSConnection(parsed_base_url.netloc, timeout=TIMEOUT)
         with closing(conn):
-            conn.putrequest(method, parsed_base_url.path + path)
-
-            # Set headers.
             headers = {
                 'Authorization': 'Bearer ' + self._get_access_token(),
-                'Transfer-Encoding': 'chunked',
                 'X-Requested-With': 'XMLHttpRequest',
             }
-            headers.update(self._extra_headers)
-            for header_name, header_value in headers.items():
-                conn.putheader(header_name, header_value)
-            conn.endheaders()
-
-            # Use chunked transfer encoding to send the data through.
-            bytes_uploaded = 0
-            while True:
-                to_send = fileobj.read(CHUNK_SIZE)
-                if not to_send:
-                    break
-                conn.send(b'%X\r\n%s\r\n' % (len(to_send), to_send))
-                bytes_uploaded += len(to_send)
-                if progress_callback is not None:
-                    should_resume = progress_callback(bytes_uploaded)
-                    if not should_resume:
-                        raise Exception('Upload aborted by client')
-            conn.send(b'0\r\n\r\n')
+            # use in-built encoding
+            conn.request(method, parsed_base_url.path + path, body=fileobj, headers=headers, encode_chunked=True)
```

Comment:
Can we add a unit test for `_upload_with_chunked_encoding` that ensures that the correct headers, for example, are sent? It could just check, for example, for the presence of the `Transfer-Encoding` header.

Reply:
where would this unit test be?

and how would it be set up? Would this function be called and then you would do a server-side check for the header?

Reply:
See https://codalab-worksheets.readthedocs.io/en/latest/Code-Overview/#unit-tests for information about unit tests.

I was initially thinking you could maybe mock the http client, but I guess now that I'm thinking about it, unit tests are probably not the best way to do this, because we are trying to just ensure that http.client works properly when passing in encode_chunked.

Can you just manually test it by running `cl upload` and ensuring that chunked uploading is being used by logging http.client calls? see httpclient_logging_patch  in https://stackoverflow.com/questions/16337511/log-all-requests-from-the-python-requests-module

Or let me know if you think of an easier way to test it manually.